### ISSUE TITLE
Reusable themes

### ISSUE THREAD

**Issue by epage** [cobalt-org/cobalt.rs#301]
Cobalt supports
- A page wrapper (loaded from `layouts`)
- Page snippets (loaded from `layouts`)
- Asses (generally not loaded from `layouts`).

The problem is this doesn't allow easily redistributed themes
- Files are spread out
- Non-theme snippets are mixed in with theme

Proposal
- Create a `snippets` directory
- Rename `layouts` to `theme`
  - `_theme.yml` for theme metadata (like asset `permalink`)
  - `*.liquid` files for page wrappers
  - `_snippets` directory that is available from `theme/` include path
  - `_data` directory that is available from `site.theme.*`
  - Everything else will be treated as an asset and copied over according to the `permalink` (default is `theme/slug.ext`)

**Comment by epage** [comment]
Do we expect our users to have a single theme for their entire site?

If they will have multiple themes, would it be per collection (pages, posts etc)?  And/or will they want multiple themes within one collection particularly pages.

Design impact: Either we have
- a single theme auto-loaded if the `_theme` folder exists
- multiple themes that are specified per collection (pages, posts, etc) or in the frontmatter

**Comment by epage** [comment]
#328 took care of the breaking changes in this.

**Comment by Geobert** [comment]
What about some overloading mechanism? A global theme, and possibility to override per collection and/or per page?

**Comment by Geobert** [comment]
And we should support Jekyll's theme out of the box, so we'll get the existing huge repository. Gutenberg has so many themes thanks to Hugo's

**Comment by epage** [comment]
>  A global theme, and possibility to override per collection and/or per page?

Yes, we should support a global theme and allow it to be overridden per-collection.  I'd like to avoid per-page unless there is a strong use case for it.  This means the directory hierarchy would be `_theme/<name>/<data>`.

> And we should support Jekyll's theme out of the box, so we'll get the existing huge repository. Gutenberg has so many themes thanks to Hugo's.

> EDIT: or enhance the Jekyll conversion to convert themes as well

The challenge is in the differences between us on Jekyll.  I agree about the conversion utility supporting themes.

**Comment by atomicangel** [comment]
I've been trying to find information that clarifies exactly how theming works with Cobalt, and there doesn't seem to be specific documentation besides "Go check out liquid and Jekyll" which doesn't point to how Cobalt implements this. I've been looking at the base Cobalt site code GH and it makes some sense but still isn't the same as full blown documentation. From what I can tell, if I just pull in a .css file and then reference it using liquid variables, that should be enough to get a site looking better than just a basic white slab with content organized into structure.

Is there any documentation that I may be overlooking? If not, would someone writing that up and doing a PR on the site code be something that is wanted/allowed/needed?

**Comment by epage** [comment]
This issue is about wanting to add theming support which would just be pre-bundled assets that you can easily switch between.

Without that, styling of the pages is just like any other HTML page.

**Comment by matbrgz** [comment]
I really like the direction laid out here, but I’d also like to propose decoupling “themes” into a true plugin system. That way, instead of bundling theme logic and assets directly into the core `_theme/<name>/…` tree, themes become first-class Cobalt plugins that:

1. **Register layouts, includes, data, and assets** via a simple plugin API
2. **Declare metadata** (name, version, dependency on Cobalt version) in their own `cobalt-plugin.toml`
3. **Live in `plugins/`** (e.g. `plugins/my-theme/`) and can be enabled/disabled per site or per collection
4. **Ship as independent crates** on crates.io, allowing theme authors to publish and iterate outside of Cobalt’s core

**Why a plugin approach?**

* **Separation of concerns**: Cobalt’s core stays lean, while theme logic (and future extensions—think shortcodes, custom filters, data transformers) live in their own repos.
* **Versioning & discovery**: Themes can follow semver independently, and a central registry (or just crates.io) becomes the place to browse community themes.
* **Per-collection & per-page overrides**: You could configure different plugins for “posts” vs. “docs” vs. “pages,” without muddying the `_theme/` folder.
* **Easier maintenance**: Bug fixes and feature additions to one theme plugin don’t require a Cobalt release.

This also aligns nicely with the idea of supporting Jekyll themes out of the box—just wrap a converted Jekyll theme as a plugin, drop it into `plugins/`, and you’re done.

Happy to help prototype a minimal plugin API if folks think this makes sense!
